### ISSUE TITLE
Add automatic class weighting support to `ModelTrainer`

### ISSUE THREAD

**Issue by gabrielhblisboa** [rafaelpadilla/3W#22]
`ModelTrainer` does not have a built-in mechanism to handle class imbalance via class weighting in the loss function.

To address class imbalance, a user must:

1.  Calculate the class weights themselves from the training data.
2.  Manually overwrite the trainer's internal loss function attribute after initialization: `trainer.criterion = nn.CrossEntropyLoss(weight=class_weights)`.

The `ModelTrainer` class could have a mechanism for accepting class_weights, and possibly even calculate them internally.
